### PR TITLE
RendezvousServer for Horovod

### LINKED ISSUE #2228: Master detects worker pod change and create/re-create RendezvousServer.

### PR DESCRIPTION
**Pull request by workingloong**
Fix #2228

### CODE REVIEW

File: elasticdl/proto/elasticdl.proto [hunk 1]
```diff
@@ -94,13 +94,25 @@ message ReportVersionRequest {
...
+}
+
 service Master {
   rpc get_task(GetTaskRequest) returns (Task);
   rpc report_evaluation_metrics(ReportEvaluationMetricsRequest)
       returns (google.protobuf.Empty);
   rpc report_task_result(ReportTaskResultRequest)
       returns (google.protobuf.Empty);
   rpc report_version(ReportVersionRequest) returns (google.protobuf.Empty);
+  rpc get_comm_rank(GetRankRequest) returns (GetRankResponse);
```
Comment (skydoorkai) [inline elasticdl/proto/elasticdl.proto]: proto change can be in another PR together with rpc implementation.

**Review by QiJune — APPROVED** [review]
Excellent!
